test/jasmine/tests/geo_test.js

```diff
@@ -688,6 +688,51 @@ describe('Test Geo layout defaults', function() {
     });
 });
 
+describe('geo visible false', function() {
+    var gd;
+
+    beforeEach(function() { gd = createGraphDiv(); });
+
+    afterEach(destroyGraphDiv);
+
+    it('should override template.layout.geo.show*', function(done) {
+        var keys = [
+            'showcoastlines',
+            'showcountries',
+            'showframe',
+            'showland',
+            'showlakes',
+            'showocean',
+            'showrivers',
+            'showsubunits'
+        ];
+
+        var layout = {
+            geo: { visible: false },
+            template: {
+                layout: {
+                    geo: {}
+                }
+            }
+        };
+
+        keys.forEach(function(k) {
+            layout.template.layout.geo[k] = true;
+        });
+
+        var data = [{ lon: [0], lat: [0], type: 'scattergeo' }];
+
+        Plotly.plot(gd, data, layout)
+        .then(function() {
+            keys.forEach(function(k) {
+                expect(gd._fullLayout.template.layout.geo[k]).toBe(false, k);
```

Comment:
At the end of the day we don't care what's in `_fullLayout.template` - only the values of `_fullLayout.geo*[k]`. Would you mind modifying the test to look at these attributes, under both of the cases we discussed: `visible: false` in the main layout as well as `visible: false` in the template?

Reply:
Good call. Fixed by https://github.com/plotly/plotly.js/pull/4483/commits/7e62ce45f14c1a1abdb05c49a631ca4f63439019.